kernel_hardening_checker/__init__.py

```diff
@@ -55,6 +55,33 @@ def detect_arch(fname: str, supported_archs: List[str]) -> Tuple[StrOrNone, str]
     return arch, 'OK'
 
 
+def detect_arch_sysctl(fname: str, supported_archs: List[str]) -> Tuple[StrOrNone, str]:
+    arch = None
+    arch_mapping = [
+        ('x86_64', 'X86_64'),
+        (['i','86'], 'X86_32'), # any of 'i?86' is x86_32
```

Comment:
How about using a Python dictionary with multiple values per key?

That would allow to avoid all this `startswith` and `endswith` shenanigans below.

Reply:
did you meant something like this?
```python 
arch_mapping = {
    'X86_64': 'x86_64',
    'ARM64': 'aarch64',
    'ARM': 'armv',        # any 'armv?' is ARM, 'armv8'+ is aarch64
    'X86_32': ['i', '86'] # any of 'i?86' is x86_32
}
```
and detecting
```python
for mapped_arch, pattern in arch_mapping.items():
    if all(subpattern in detected_arch for subpattern in pattern):
        arch = mapped_arch
```

Reply:
Yes, it looks better, please see the comment below.